### PR TITLE
Java parsing: Allow access to new array with an initializer.

### PR DESCRIPTION
**Pull request by mjambon**
Fixes https://github.com/returntocorp/semgrep/issues/906

My understanding is that this construct falls outside of the official grammar specification, but it is implemented in javac and we have to support it.

### CODE REVIEW

File: tests/java/parsing/NewArrayAccess.java [hunk 1]
```diff
@@ -0,0 +1,9 @@
+// NewArrayAccess.java
+//
+// Compare to: NewArrayMultidim.java
+//
+public class NewArrayAccess {
+    static String get(int index) {
+        return new String[] { "Courier", "Impact" }[index];
+    }
+}
```
Comment (mjambon) [inline tests/java/parsing/NewArrayAccess.java:9]: this was failing

File: tests/java/parsing/NewArrayMultidim.java [hunk 2]
```diff
@@ -0,0 +1,6 @@
+// NewArrayMultidim.java
+public class NewArrayMultidim {
+    static String[][] get(int index) {
+        return new String[3][index];
+    }
+}
```
Comment (mjambon) [inline tests/java/parsing/NewArrayMultidim.java:6]: this was already working

File: lang_java/parsing/parser_java.mly [hunk 3]
```diff
@@ -434,6 +434,8 @@ primary_no_new_array:
  | class_literal       { $1 }
  /*(* javaext: ? *)*/
  | method_reference { $1 }
+ /*(* javaext: ? *) */
+ | array_creation_expression_with_initializer { $1 }
```
Comment (nbrahms) [inline lang_java/parsing/parser_java.mly:438]: This probably deserves a descriptive comment, especially as it's under the `primary_no_new_array` rule.

Comment (mjambon) [reply lang_java/parsing/parser_java.mly:438]: I agree but I wasn't sure what to do about comments. I don't see many comments in there and I appreciate the ability to see just the structure of the grammar without clutter. I'll describe the problem just before the new rule, I think that's best.

Comment (mjambon) [reply lang_java/parsing/parser_java.mly:442]: added comments and merging.

**Comment by aryx** [comment]
Fantastic! And there was no additional shift/reduce conflicts?

**Pull request merged**